File: output/CodeGen/avatar/Java/Python/atcoder_ABC122_C.py
class atcoder_ABC122_C {
    final static long mod = 1000000007

    def main(args):
        sc = Scanner(System.in)
        n = sc.nextInt()
        q = sc.nextInt()
        c = sc.next().strip().toCharArray()
        range = [[sc.nextInt(), sc.nextInt()]]
        for i in range(q):
            range.append([sc.nextInt(), sc.nextInt()])
        frag = [0] * (n + 1)
        rui = [0] * (n + 1)
        for i in range(2, n + 1):
            if c[i - 2] == 'A' and c[i - 1] == 'C':
                frag[i] += 1
            rui[i] = rui[i - 1] + frag[i]
        for i in range(q):
            left, right = range[i][0], range[i][1]
            print(rui[right] - rui[left])
        return


    main()
}

File: output/CodeLlama/avatar/Python/Java/atcoder_ABC122_C.java
import java.util.Scanner;
import java.util.stream.IntStream;

public class atcoder_ABC122_C {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int q = sc.nextInt();
        String s = sc.next();
        int[][] problems = new int[q][2];
        for (int i = 0; i < q; i++) {
            problems[i][0] = sc.nextInt();
            problems[i][1] = sc.nextInt();
        }
        int[] cumsum = IntStream.range(0, n).map(i -> {
            if (i == 0) return 0;
            if (s.substring(i - 1, i + 1).equals("AC")) return 1;
            return 0;
        }).toArray();
        for (int[] problem : problems) {
            System.out.println(cumsum[problem[1] - 1] - cumsum[problem[0] - 1]);
        }
    }
}